### PR TITLE
Updated TextView component to implement AutoPropertyDiffable

### PR DESCRIPTION
**Pull request by aducret**
## Screencast
pr

### CODE REVIEW

**Comment by aducret** [comment]
[Hacer que TextView sea diffable en Portal](https://trello.com/c/dPsun3LY/295-hacer-que-textview-sea-diffable-en-portal)

File: Portal/View/Components/TextField.swift [hunk 1]
```diff
@@ -119,12 +119,12 @@ internal struct TextFieldChangeSet<MessageType> {
     
     static func fullChangeSet(
         properties: TextFieldProperties<MessageType>,
-        styleSheet: StyleSheet<TextFieldStyleSheet>,
+        style: StyleSheet<TextFieldStyleSheet>,
```
Comment (pablog) [inline Portal/View/Components/TextField.swift:122]: Why did you rename this?

File: Portal/View/Portal.swift [hunk 2]
```diff
@@ -12,6 +12,10 @@ public struct Font: AutoEquatable {
     
     let name: String
     
+    public init(name: String) {
```
Comment (pablog) [inline Portal/View/Portal.swift:15]: Isn't this `init` generated automatically?

File: Portal/View/Components/TextField.swift (hunk 1 above)
Comment (aducret) [reply Portal/View/Components/TextField.swift:122]: To be consistent with others components like Button

File: Portal/View/Portal.swift (hunk 2 above)
Comment (aducret) [reply Portal/View/Portal.swift:15]: yep, but it's internal by default and now the only way to create a Font is with this init (as Portal user).

**Review by pablog — APPROVED** [review]

**Comment by aducret** [comment]
@pablog I added a little change, if you can please check it.

I think `TextType` should be optional in TextViewChangeSet to represent the state when the textType doesn't change.

**Review by pablog — APPROVED** [review]

File: Portal/View/SingleProperty.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+//
+//  SingleProperties.swift
+//  Portal
+//
+//  Created by Argentino Ducret on 8/29/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+struct SingleProperty<T> {
```
Comment (guidomb) [inline Portal/View/SingleProperty.swift]: I think it would be better to have

```swift
enum PropertyChange<T> {

  case change(to: T)
  case noChange

}
```

and update `TextViewChageSet`'s `textType` property to be of type `PropertyChange`.

File: PortalTests/TextViewRenderSpec.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                    )
+                    
+                    let textView = UITextView()
+                    let _: Render<String> = textView.apply(changeSet: changeSet, layoutEngine: layoutEngine)
+                    
+                    expect(textView.attributedText.string).to(equal("This is a textView!"))
+                }
+                
+                
+                // TODO: Check NSAttributedString's attributes
```
Comment (guidomb) [inline PortalTests/TextViewRenderSpec.swift:69]: What about this?

**Review by guidomb — CHANGES_REQUESTED** [review]

File: PortalTests/TextViewRenderSpec.swift (hunk 4 above)
Comment (aducret) [reply PortalTests/TextViewRenderSpec.swift:69]: When we apply a changeSet to a NSAttributedString a `NSOrignalFont` key is added but if I instantiate a new AttributedString with the same configuration it doesn't have this key.

We should research a little about why the NSAttributedString has in some contexts an attribute called `NSOriginalFont`. 

I prefer keep advancing with other components and then check this.

**Review by guidomb — APPROVED** [review]
